### ISSUE TITLE
Kernel breaks  when doing backward pass with .permute with the -1 option

### ISSUE THREAD

**Issue by manuelsh** [pytorch/pytorch#9172]
## Issue description

Permuting the axis of a tensor using the `-1` option breaks the kernel (tested in Jupyter) when doing the backward pass. The kernel just restart, there is no error provided.

## Code example

This code:
```
import torch
from torch.autograd import Variable
a = Variable(torch.rand(2,3,2), requires_grad = True)
a = a.permute(1,0,-1)
a = a.sum()
a.backward()
```
automatically breaks the Kernel of a Jupyter Notebook (run in JupyterLab).

Note that the following code:

```
import torch
from torch.autograd import Variable
a = Variable(torch.rand(2,3,2), requires_grad = True)
a = a.permute(1,0,2)
a = a.sum()
a.backward()
```
worked well.

## System Info
PyTorch version: 0.3.1
Is debug build: No
CUDA used to build PyTorch: 9.0.176

OS: Ubuntu 16.04.3 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609
CMake version: version 3.5.1

Python version: 3.5
Is CUDA available: Yes
CUDA runtime version: 9.0.176
GPU models and configuration: GPU 0: Tesla K80
Nvidia driver version: 384.111
cuDNN version: Probably one of the following:
/usr/local/cuda-8.0/lib64/libcudnn.so.6.0.21
/usr/local/cuda-8.0/lib64/libcudnn_static.a
/usr/local/cuda-9.0/lib64/libcudnn.so.7.0.5
/usr/local/cuda-9.0/lib64/libcudnn_static.a
/usr/local/lib/python3.5/dist-packages/torch/lib/libcudnn-7b07b0f1.so.7.0.5

Versions of relevant libraries:
[pip3] msgpack-numpy (0.4.1)
[pip3] numpy (1.14.0)
[pip3] torch (0.3.1)
[pip3] torchvision (0.2.0)
[conda] Could not collect

**Comment by SsnL** [comment]
What about in command line? The code works on master build with command line python.

**Comment by manuelsh** [comment]
Not happening in 0.4.0. 

Happens only in 0.3.1.

Closing the issue. Thanks.
